src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,129 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, inject, OnInit } from "@angular/core";
+import AppValues from "src/app/common/app.values";
+import { Clipboard } from "@angular/cdk/clipboard";
+import { QueryExplainerService } from "./query-explainer.service";
+import { BaseComponent } from "src/app/common/base.component";
+import { MaybeNull, MaybeUndefined } from "src/app/common/app.types";
+import ProjectLinks from "src/app/project-links";
+import { catchError, forkJoin, Observable, of, switchMap, tap } from "rxjs";
+import { QueryExplainerResponse, VerifyQueryResponse } from "./query-explainer.types";
+import { HttpErrorResponse } from "@angular/common/http";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { DataRow, DataSchemaField } from "src/app/interface/dataset.interface";
+import { extractSchemaFieldsFromData } from "src/app/common/table.helper";
+import { BlockService } from "src/app/dataset-block/metadata-block/block.service";
+
+@Component({
+    selector: "app-query-explainer",
+    templateUrl: "./query-explainer.component.html",
+    styleUrls: ["./query-explainer.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryExplainerComponent extends BaseComponent implements OnInit {
+    private clipboard = inject(Clipboard);
+    private queryExplainerService = inject(QueryExplainerService);
+    private cdr = inject(ChangeDetectorRef);
+    private blockService = inject(BlockService);
+
+    private sqlQuery: MaybeNull<string>;
+    public readonly DATE_FORMAT = AppValues.DISPLAY_FLOW_DATE_FORMAT;
+
+    public sqlQueryExplainer: QueryExplainerResponse;
+    public sqlQueryVerify$: Observable<VerifyQueryResponse>;
+    public verifyResponse: VerifyQueryResponse;
```

Comment:
So input `sqlQuery` may be null, but the resulting data can't? I'm not sure this typing is accurate.